syntax = "proto3";
package auth;

enum Country {
  UKRAINE = 0;
  USA = 1;
  UK = 2;
  POLAND = 3;   
  JAPAN = 4;   
  AUSTRALIA = 5;
}

message Address {
  optional string street = 1;
  string city = 2;
  Country country = 3;
}

message User {
  string email = 1;
  optional string first_name = 2;
  optional string last_name = 3;
  optional Address address = 4;
}

// Request message containing email to log in
message LoginRequest {
  string email = 1;
}

// Request message containing User to register
message RegisterRequest {
  User user = 1;
}

// Request message containing page size to get page of Users
message GetUsersRequest {
  int32 page_size = 1;
  int32 page_number = 2;
}

// Response message containing list of users and total number of pages
message GetUsersResponse {
  repeated User users = 1;
  int32 total_pages = 2;
}

// Auth service definition
service AuthService {
    // Login User with email
    rpc Login(LoginRequest) returns (User) {}
    // Register User with User object
    rpc Register(RegisterRequest) returns (User) {}
    // Get page of Users
    rpc GetUsers(GetUsersRequest) returns (GetUsersResponse) {}
}